### PR TITLE
:tada: vulnerability API added

### PR DESCRIPTION
**Pull request by tonnyhideyori**

### CODE REVIEW

File: implementation.py [hunk 1]
```diff
@@ -86,3 +86,110 @@ def update_project(self, uuid, name=None, classifier=None):
             return (f"Project with specified name already exists", response.status_code)
         else:
             return (f"Unable to update the project", response.status_code)
+    #This section is all about vulnerabilities
+    
+    def get_all_vulnerabilities(self):
+        response = self.session.get(self.apicall +f"/v1/vulnerability")
```
Comment (manuel-sommer) [inline implementation.py:92]: Thank you for contributing @tonnyhideyori 
Could you please add paging here (see def list_projects)

File: implementation.py [hunk 2]
```diff
@@ -86,3 +86,110 @@ def update_project(self, uuid, name=None, classifier=None):
...
+            else:
+                return (f"The vulnerability could not be found ", response.status_code)
+    
+    def get_component_vulnerability(self,uuid,supressed=False):
+        """ 
+        Returns a list of all vulnerabilities for a specific component.
+        uuid:
+        supprressed: optionally includes supressed vulnerabilities
+        """
+        response = self.session.get(self.apicall +f"/v1/vulnerability/component/{uuid}?supressed={supressed}")
```
Comment (manuel-sommer) [inline implementation.py:119]: Could you please add paging here (see def list_projects)

File: implementation.py [hunk 3]
```diff
@@ -86,3 +86,110 @@ def update_project(self, uuid, name=None, classifier=None):
...
+            else:
+                return (f"The component could not be found", response.status_code)
+    
+    def get_project_vlnerability(self, uuid,supressed=False):
+        """ 
+        Returns a list of all vulnerabilities for a specific project.
+        uuid:
+        supprressed: optionally includes supressed vulnerabilities(boolean)
+        """
+        response = self.session.get(self.apicall +f"/v1/vulnerability/project/{uuid}?supressed={supressed}")
```
Comment (manuel-sommer) [inline implementation.py:136]: Could you please add paging here (see def list_projects)

File: implementation.py [hunk 4]
```diff
@@ -86,3 +86,110 @@ def update_project(self, uuid, name=None, classifier=None):
...
+            else:
+                return (f"The vulnerability could not be found ", response.status_code)
+
+    def get_affected_project(self,source,vuln):
+        """ 
+        Returns a list of all projects affected by a specific vulnerability
+        source:
+        vuln:
+        """
+        response = self.session.get(self.apicall +f"/v1/vulnerability/source/{source}/vuln/{vuln}/projects")
```
Comment (manuel-sommer) [inline implementation.py:167]: Could you please add paging here (see def list_projects)

File: implementation.py [hunk 5]
```diff
@@ -86,3 +86,110 @@ def update_project(self, uuid, name=None, classifier=None):
...
+            return response.json()
+        else:
+            if response.status_code==401:
+                return (f"Unauthorized",response.status_code)
+            elif response.status_code==403:
+                return (f"Access to the specified component is forbidden ", response.status_code)
+            else:
+                return (f"The component could not be found", response.status_code)
+    
+    def get_project_vlnerability(self, uuid,supressed=False):
```
Comment (manuel-sommer) [inline implementation.py:130]: typo

File: implementation.py [hunk 6]
```diff
@@ -86,3 +86,110 @@ def update_project(self, uuid, name=None, classifier=None):
...
+            else:
+                return (f"The vulnerability could not be found ", response.status_code)
+    #TODO: POST,POST /v1/vulnerability
+    #TODO: DELETE, POST  /v1/vulnerability/source/{source}/vuln/{vulnId}/component/{component}, DELETE, POST /v1/vulnerability/{uuid}/component/{component}
+    
+    #this section is all about findings
+    
+    def get_project_vlnerability(self, uuid,supressed=False):
+        """ 
+        Returns a list of all findings for a specific project.
```
Comment (manuel-sommer) [inline implementation.py:182]: This is a duplicate method.

**Review by manuel-sommer — COMMENTED** [review]
Hi @tonnyhideyori , could you please add the following comments?
Thank you for contributing.

File: implementation.py (hunk 6 above)
Comment (tonnyhideyori) [reply implementation.py:182]: done in new commit

File: implementation.py (hunk 5 above)
Comment (tonnyhideyori) [reply implementation.py:130]: done in new commit

**Comment by manuel-sommer** [comment]
You mean you fixed all these in the new PR? Please fix the comments next time directly within this PR.

**Pull request merged**